src/lib.rs

```diff
@@ -1054,6 +1070,55 @@ where
     }
 }
 
+// borsh is byte-size little-endian de-needs-external-schema no-bit-compression serde
+// current ser/de for it is not optimal impl because const math is not stable nor primitives has bits traits
+#[cfg(feature = "borsh")]
+impl<T: BorshSerialize, const BITS: usize> BorshSerialize for UInt<T, BITS>
+where
+    Self: Number,
+{
+    fn serialize<W: borsh::io::Write>(&self, writer: &mut W) -> borsh::io::Result<()> {
+        self.value.serialize(writer)
+    }
+}
+
+#[cfg(feature = "borsh")]
+impl<
+        T: BorshDeserialize + core::cmp::PartialOrd<<UInt<T, BITS> as Number>::UnderlyingType>,
+        const BITS: usize,
+    > BorshDeserialize for UInt<T, BITS>
+where
+    Self: Number,
+{
+    fn deserialize_reader<R: borsh::io::Read>(reader: &mut R) -> borsh::io::Result<Self> {
+        let value = T::deserialize_reader(reader)?;
+        if value >= Self::MIN.value() && value <= Self::MAX.value() {
+            Ok(Self { value })
+        } else {
+            Err(borsh::io::Error::new(
+                borsh::io::ErrorKind::InvalidData,
+                "Value out of range",
+            ))
+        }
+    }
+}
+
+#[cfg(feature = "borsh")]
+impl<T, const BITS: usize> BorshSchema for UInt<T, BITS> {
+    fn add_definitions_recursively(
+        definitions: &mut BTreeMap<borsh::schema::Declaration, borsh::schema::Definition>,
+    ) {
+        definitions.insert(
+            ["u", &BITS.to_string()].concat().into(),
+            borsh::schema::Definition::Primitive(((BITS + 7) / 8) as u8),
```

Comment:
This is the number of bytes, but it doesn't seem to match what's actually being stored above? Is that a problem?